MaterialLib/MPL/CheckMaterialSpatialDistributionMap.h

```diff
@@ -0,0 +1,52 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2020, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "MaterialSpatialDistributionMap.h"
+#include "Medium.h"
+#include "MeshLib/Mesh.h"
+#include "Phase.h"
+
+namespace MaterialPropertyLib
+{
+template <typename ContainerMedium,
+          typename ContainerSolid,
+          typename ContainerLiquid>
+void checkMaterialSpatialDistributionMap(
```

Comment:
The implementation should go into cpp file. Forward decls in the header.
The translation units where this file is included (the processes), are already expensive to compile. Separating the implementation helps with decoupling.